docs/user/gcp/install_upi.md

```diff
@@ -292,60 +301,49 @@ EOF
 - `region`: the region to deploy the cluster into (for example us-east1)
 - `cluster_network`: the URI to the cluster network
 - `network_cidr`: the CIDR of the vpc network (for example 10.0.0.0/16)
-- `master_nat_ip`: the ip address of the master nat (for example 34.94.100.1)
-- `worker_nat_ip`: the ip address of the worker nat (for example 34.94.200.1)
 
 Create the deployment using gcloud.
 
 ```sh
 gcloud deployment-manager deployments create ${INFRA_ID}-security --config 03_security.yaml
 ```
 
+## Configure security variables
+```sh
+export MASTER_SERVICE_ACCOUNT=$(gcloud iam service-accounts list | grep "^${INFRA_ID}-master-node " | awk '{print $2}')
+export WORKER_SERVICE_ACCOUNT=$(gcloud iam service-accounts list | grep "^${INFRA_ID}-worker-node " | awk '{print $2}')
```

Comment:
Why are we using list ? What's the behavior of gcloud when paging and why not direct describe or filters?

Reply:
The goal is to discover the `email` for the service accounts, which we would need to do any other action. We could either guess it based on `${INFRA_ID}-m@${PROJECT}.iam.gserviceaccount.com` or discover it using `gcloud iam service-accounts list --filter "email~${INFRA_ID}-m@" --format json | jq -r ".[0].email"` The first option would require no connections to gcloud, but might wrongly assume the `.iam.gserviceaccount.com` if it were to change. The second option requires connections to gcloud. Which do you prefer?

Reply:
> The goal is to discover the email for the service accounts
why is there is need to discover? didn't we create the service accounts?

Reply:
We did create them, but we didn't specify `iam.gserviceaccount.com` at any point in creation. That is automatically added somewhere in the gcloud API. So really we are just discovering that. So the real question is how confident are we that will not change base on location, cloud partition, future changes, etc.

Reply:
> https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts#ServiceAccount

based on how _we_ created the service account, the displayName could be used to not _discover_

also does GDM not return information about the created resources?

Reply:
The displayName is not a field we can use as an identifier in any of the gcloud commands. Also, it's possible for two keys to have the same displayName as they don't appear to be unique (something I only discovered today).

We aren't using GDM output anywhere else to fill variables.

Reply:
> We aren't using GDM output anywhere else to fill variables.

and?

Reply:
We might be able to grab it from `gcloud deployment-manager deployments describe`, but that isn't significantly better than `gcloud deployment-manager deployments describe`.

Reply:
> The displayName is not a field we can use as an identifier in any of the gcloud commands. 

that's true but atleast filtering on that prevents use missing it al together if gcloud list is not returning multiple pages.. 

> Also, it's possible for two keys to have the same displayName as they don't appear to be unique (something I only discovered today).

across projects is possible, but in the same project, i don't think so.

> We might be able to grab it from gcloud deployment-manager deployments describe, but that isn't significantly better than gcloud deployment-manager deployments describe.

we would know exactly what service account was created from the GDM, and not have to guess...

Reply:
Addressed in 9f2ff90605c4d291eb28ccaa065ce01872988a61